app/src/main/java/com/gh4a/utils/RxUtils.java

```diff
@@ -0,0 +1,185 @@
+package com.gh4a.utils;
+
+import android.app.Dialog;
+import android.os.Bundle;
+import android.support.annotation.StringRes;
+import android.support.design.widget.CoordinatorLayout;
+import android.support.design.widget.Snackbar;
+import android.support.v4.app.DialogFragment;
+import android.support.v4.app.FragmentActivity;
+
+import com.gh4a.ApiRequestException;
+import com.gh4a.BaseActivity;
+import com.gh4a.R;
+import com.meisolsson.githubsdk.model.Page;
+import com.meisolsson.githubsdk.model.SearchPage;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+
+import io.reactivex.Single;
+import io.reactivex.SingleSource;
+import io.reactivex.SingleTransformer;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.functions.Function;
+import io.reactivex.functions.Predicate;
+import io.reactivex.processors.PublishProcessor;
+import io.reactivex.schedulers.Schedulers;
+import retrofit2.Response;
+
+public class RxUtils {
+    public static <T> Single<T> throwOnFailure(Single<Response<T>> upstream) {
```

Comment:
I think that this method is useless. Can't we simply call `.map(ApiHelpers::throwOnFailure) `instead of `.compose(RxUtils::throwOnFailure)`?

Reply:
Will check this.